bundles/edu.kit.kastel.sdq.eclipse.grading.api/src/edu/kit/kastel/sdq/eclipse/grading/api/client/IArtemisClient.java

```diff
@@ -97,4 +101,22 @@ void downloadExerciseAndSubmission(IExercise exercise, ISubmission submission, F
 	 *                                or if authentication fails.
 	 */
 	Optional<ILockResult> startNextAssessment(IExercise exerciseID, int correctionRound) throws ArtemisClientException;
+
+	void downloadExercise(IExercise exercise, File dir, IProjectFileNamingStrategy namingStrategy, String repoUrl) throws ArtemisClientException;
```

Comment:
I think repoUrl might not be important here. Exercise should contain url for user?